### ISSUE TITLE
Compress single file slow using GZIP Apple Silicon M1

### ISSUE THREAD

**Issue by jojeswe** [aonez/Keka#777]
Im using an Apple Mac Mini M1 together with Keka v1.2.8 When compressing one 4GB single file using GZIP it takes 4-5 minutes. If I compress the same file together with a small 1KB file archiving and compressing takes altogether 20 seconds. I have "Always tarball" option set in both cases.

When compressing two files I can see the following two processes
kekapigz
kekatar

When compressing one file I can see the following process.
keka7z

Based on the big  difference in performance can I change which binary is used for compressing single files to GZIP?

/Johan

**Comment by aonez** [comment]
Thanks for the feedback @jojeswe!

You said you have "Always tarball" selected? With this option it should always use `TAR` + `PIGZ`. You can you check the internal value of this option using the Terminal, it should return 1:

```
defaults read com.aone.keka ForceTarballOnCompressionOnly
```

![Screenshot 2021-01-28 at 08 54 58](https://user-images.githubusercontent.com/480163/106107142-d8edcc00-6146-11eb-89ca-b56ea4b22db6.png)

You can also uncheck and check the option. If it still uses `7Z` please let me know.
I should probably switch to `PIGZ` instead of `7Z`, but the later adds the split functionality.

**Comment by jojeswe** [comment]
Hi, thanks @aonez for the response. I have unchecked/checked the "Always tarball..." but when compressing this large file it still only creates a .gz file. Testing further, I created a new 4GB dummy file, when compressing this new file it works as expected.

The "defaults read com.aone.keka ForceTarballOnCompressionOnly" returns 1.

The original file has been modified before compressing using the tail command and I have also changed the encoding of the file using iconv before compressing. What in my original file could trigger this behaviour in Keka? 



/Johan

**Comment by aonez** [comment]
Maybe you can `Create a debug session` (`Help` -> `Debug`), start the compression that uses `7Z` stop it after a few seconds and send me the log file? Also how are you triggering the compression? Dropping in the window, Finder extension, service...

> The original file has been modified before compressing using the tail command and I have also changed the encoding of the file using iconv before compressing. What in my original file could trigger this behaviour in Keka?

It shouldn't make any difference on compression... But let's check the log.

**Comment by jojeswe** [comment]
I also sent the same file to another Macbook (Intel + Big Sur + Keka 1.2.9) and the file does not compress to tar.gz, it can only be compressed to .gz and takes for ever.

**Comment by aonez** [comment]
Is that file private? Maybe you can share it with me? info@keka.io

**Comment by jojeswe** [comment]
I can not share the file since it contains corporate information. Attached you will fins the log.
[Keka_2021-01-28-102036907.log](https://github.com/aonez/Keka/files/5886099/Keka_2021-01-28-102036907.log)

**Comment by jojeswe** [comment]
@aonez , I have sent you a small file that has the same problem. Let me know if you can repeat the issue.

**Comment by aonez** [comment]
Still can't reproduce the issue with that file but if the file dropped is a TAR it's compressed using 7Z, so not sure what exactly outputs `tail` + `iconv` but probably the resulting file is handled as a TAR.

If you drop that file to Keka the default operation is extracting it?

I will use PIGZ instead of 7Z if split is not used, seeing that difference in performance.

Thanks a lot for the feedback @jojeswe!

**Comment by jojeswe** [comment]
The original file is just an uncompressed CSV file from a windows server. I then applied Trail to remove the last row of the file and finally I changed the encoding. I then took the resulting CSV file and compressed it using Keka, this file always gets slowly compressed to *.gz (no TAR) using the 7Z binary. Any other file I can compress to *tar.gz

The PIGZ binary seems much faster then 7Z so if that could be used instead that would be better.

**Comment by aonez** [comment]
Can you check what signature that CSV returns:

`xxd -l 20 YUORFILE.csv`

Maybe this gives me some light...

**Comment by jojeswe** [comment]
@aonez I sent you the signature. The signature is the same for the small file I sent you and the corporate 4GB file.

**Comment by aonez** [comment]
Johan,

As a last followup, the file is indeed a TAR file. Forgot that TAR signature is at offset `0x101`, you can check it in the big file too:

```
xxd -l 5 -s 0x101 YUORFILE.CSV
```

This will give you `7573 7461 72` -> `ustar`, the TAR signature.

![Screenshot 2021-02-01 at 09 27 23](https://user-images.githubusercontent.com/480163/106433191-1f596880-6470-11eb-954a-753e1b5626ce.png)

I'll temporarily fix this by only detecting TAR files when compressing if they have TAR extension.

But for the future [using PIGZ will be the way to go](#782). Thanks again for all the feedback!

**Comment by jojeswe** [comment]
Hi, Yes that's correct. I get the TAR signature in the big tile too.

/J

**Comment by aonez** [comment]
This will be fixed in the upcoming 1.2.11.

**Comment by aonez** [comment]
@jojeswe this was fixed in version 1.2.10 (past week). Sorry for the misinformation. Reopen if you find this again.

I'll try to implement the use of PIGZ instead of 7Z for non tarball operations in #782.
